### PR TITLE
Admin Group Page

### PR DESCRIPTION
**Pull request by jmf373**
## Overview

Added the admin-side designs to the group page:
 - Add poll button
 - Export button
 - View all button
Added the empty-state views for students and admins
Passed in the user role in each of the entry points to the group page

  ## Changes Made

 - Changed `activity_polls_date.xml` to include empty state view and admin views
 - Changed `GroupFragment` and `GroupRecyclerView` to pass user role to the group page
 - Changed `PollsDateActivity` to display/hide empty state and admin views as needed

  ## Screenshots

  <details>

    <summary>Admin View</summary>

Screen Shot 2019-10-16 at 6 16 07 PM

    <summary>Student Empty State</summary>

Screen Shot 2019-10-16 at 6 23 11 PM

    <summary>Admin Empty State</summary>

Screen Shot 2019-10-16 at 6 47 23 PM

  </details>

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/pollo/GroupFragment.kt [hunk 1]
```diff
@@ -49,10 +49,16 @@ class GroupFragment(val callback: OnMoreButtonPressedListener) : Fragment() {
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View {
         val rootView = inflater!!.inflate(R.layout.fragment_main, container, false)
+        val role: Int
+        if (arguments?.getString(GroupFragment.GROUP_ROLE) == "admin") {
+            role = 1
+        } else {
+            role = 0
+        }
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/pollo/GroupFragment.kt]: let's reduce this to one line: https://kotlinlang.org/docs/reference/control-flow.html

File: app/src/main/java/com/cornellappdev/android/pollo/GroupRecyclerView.kt [hunk 2]
```diff
@@ -15,7 +14,7 @@ import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
-class GroupRecyclerAdapter(private val groups: ArrayList<Group>, val callback: GroupFragment.OnMoreButtonPressedListener?) : androidx.recyclerview.widget.RecyclerView.Adapter<GroupRecyclerAdapter.ViewHolder>() {
+class GroupRecyclerAdapter(private val groups: ArrayList<Group>, val callback: GroupFragment.OnMoreButtonPressedListener?, val role: Int) : androidx.recyclerview.widget.RecyclerView.Adapter<GroupRecyclerAdapter.ViewHolder>() {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/pollo/GroupRecyclerView.kt]:
```
class GroupRecyclerAdapter (
  private val groups: ArrayList<Group>, 
  val callback: GroupFragment.OnMoreButtonPressedListener?, 
  val role: Int
) : androidx.recyclerview.widget.RecyclerView.Adapter<GroupRecyclerAdapter.ViewHolder>() {
```

File: app/src/main/java/com/cornellappdev/android/pollo/GroupFragment.kt (hunk 1 above)
Comment (ningning621) [reply app/src/main/java/com/cornellappdev/android/pollo/GroupFragment.kt]:
```
role = if (arguments?.getString(GroupFragment.GROUP_ROLE) == "admin") 1 else 0
```

File: app/src/main/res/layout/activity_polls_date.xml [hunk 3]
```diff
@@ -5,66 +5,154 @@
...
+
+                <ImageButton
+                    android:background="@color/black"
+                    android:contentDescription="@string/go_back_content_description"
+                    android:layout_alignParentStart="true"
+                    android:layout_centerVertical="true"
+                    android:layout_height="32dp"
+                    android:layout_marginStart="15dp"
+                    android:layout_width="32dp"
+                    android:onClick="goBack"
```
Comment (ningning621) [inline app/src/main/res/layout/activity_polls_date.xml]: tbh i'm not a fan of setting the `onClick()` in the xml. there shouldn't be any logic in the xml files

**Comment by ningning621** [comment]
hmm i'm not a fan of using an integer flag to denote user vs. admin bc devs who don't work on this codebase may not know what it means unless they hunt thru the code. if there's only two roles, is it possible to use a boolean flag instead, or a enum?

**Comment by jmf373** [comment]
Somehow didn't realized `Enum`'s were serializable... Fixed that up and the other small changes

**Pull request merged**